Data/DataModel/Puzzle.cs

```diff
@@ -43,34 +46,70 @@ public class Puzzle
         /// </summary>
         public int MinPrerequisiteCount { get; set; } = 0;
 
-        [DataType(DataType.Url)]
-        public string PuzzleUrlString { get; set; }
+        /// <summary>
+        /// All of the content files associated with this puzzle
+        /// </summary>
+        public virtual ICollection<ContentFile> Contents { get; set; }
 
+        /// <summary>
+        /// File for the main puzzle (typically a PDF containing the puzzle)
+        /// </summary>
         [NotMapped]
-        public Uri PuzzleUrl
+        public ContentFile PuzzlePDF
```

Comment:
What if it's not a PDF? Why be specific about that?

Reply:
Ditto AnswerPDF of course.

Reply:
Will change to PuzzleFile/AnswerFile